Core/Operator/Instance.cs

```diff
@@ -74,28 +74,28 @@ where typeof(IInputSlot).IsAssignableFrom(field.FieldType)
 
         public bool AddConnection(Symbol.Connection connection, int multiInputIndex)
         {
-            var (_, sourceSlot, _, targetSlot) = GetInstancesForConnection(connection);
-            if (targetSlot != null)
-            {
-                targetSlot.AddConnection(sourceSlot, multiInputIndex);
-                return true;
-            }
+            var foundInstance = TryGetInstancesForConnection(connection, out _, out var sourceSlot, out _, out var targetSlot);
```

Comment:
This code looks so weird... 
I can't help but to think there must be a better way.

Reply:
HARD agree on that one. I think it warrants being split into two methods - one for add and one for remove. I just translated what was there more or less

Reply:
I did it